source/service/k3s.rst

```diff
@@ -0,0 +1,952 @@
+
+``ONEAPP_K8S_LOADBALANCER_RANGE`` is the simplest way to configure the LoadBalancer type of services. The value must be a range (or multiple ranges by adding more params with numbered suffix), e.g.:
+
+.. code::
+
+    192.168.100.100-192.168.100.200
+
+More info is in the :ref:`LoadBalancer section <k3s_loadbalancer>`.
+
+.. note::
+
+   ``ONEAPP_K8S_LOADBALANCER_RANGE`` and ``ONEAPP_K8S_LOADBALANCER_CONFIG`` are applicable only if MetalLB is used.
+
+.. important::
+
+   Handle carefully the K3s cluster token (``k8s_token`` in :ref:`/etc/one-appliance/config <one_service_logs>`) as it **never** expires.
+
+.. note::
+
+    There is an option to setup master node as a `control-plane <https://kubernetes.io/docs/concepts/overview/components/#control-plane-components>`_ only. In such scenario the master node will not schedule any pods on itself and it will run user pods on worker nodes only. Therefore single-node master with control-plane only does not make sense and it will always need other (worker) nodes.
+
+    If you wish to run the master node in such way then set the ``ONEAPP_K8S_TAINTED_MASTER=YES``.
+
+.. _k3s_worker_node:
+
+Worker Node
+-----------
+
+New nodes can join already running K3s cluster, if they are provided with following contextualization parameters:
+
+* ``ONEAPP_K8S_ADDRESS`` - master node's IP address
+* ``ONEAPP_K8S_TOKEN``
+
+.. important::
+
+   For the worker nodes, the ``ONEAPP_K8S_ADDRESS`` must be an exact IP address of the master node on which the K3s advertises its API!
+
+   Both parameters must be defined together to take an effect (with correct values). Otherwise a node will be deployed as an independent master of a single-node cluster.
+
+You can find all values for these parameters on the master node in the :ref:`/etc/one-appliance/config <one_service_logs>` (actual values will differ from our :ref:`example <report_file_example>`):
+
+* ``k8s_master``
+* ``k8s_token``
+
+The example use of these parameters can be seen on the figure:
+
+|image-worker-values|
+
+When you check the status of the nodes in the cluster, you should see one master and all the other joined nodes.
+
+Example:
+
+.. prompt:: text [master]# auto
+
+   [master]# kubectl get nodes
+   NAME                        STATUS   ROLES                  AGE    VERSION
+   onek3s-ip-192-168-150-101   Ready    control-plane,master   150m   v1.21.5+k3s2
+   onek3s-ip-192-168-150-102   Ready    <none>                 63s    v1.21.5+k3s2
+
+.. _k3s_oneflow:
+
+OneFlow Integration
+-------------------
+
+The K3s appliance (optionally) integrates with the OpenNebula `OneFlow service <https://docs.opennebula.io/stable/management_and_operations/multivm_service_management/appflow_use_cli.html>`_ to enable one-click multi-node cluster deployment. A template for OneFlow is not part of the imported Marketplace appliance, it must be created in each OpenNebula manually. The template can be created over the Sunstone wizard, or by providing a JSON specification.
```

Comment:
I would say here that the Service template exists only for OpenNebula 6.0 and newer (what it is said in the subsequent note) and can be imported from the Marketplace. For previous releases it is not available, and in this section it is explained how to build a service template for k3s deployment.